lectures/5-commitments.tex

```diff
@@ -0,0 +1,255 @@
+        
+        $$
+        \mathsf{HideAdv}[\mathcal{A}, \Pi_{\text{commitment}}] := \left| \text{Pr}[b=\hat{b}] - \frac{1}{2} \right|
+        $$
+        
+        We say that the commitment scheme $\Pi_{\text{commitment}}$ is \textit{hiding} if for any adversary, the aforementioned advantage is negligible.
+        
+        \item \textit{Binding: } prover could not find another message $m_1$ and open the commitment $c$ without revealing the commited message $m$.
+
+    \end{enumerate}
+\end{definition}
+
+\subsubsection{Hash-based commitments}
+
+As the name implies, we are using a cryptographic hash function \(H\) in such scheme.
+
+\begin{enumerate}
+    \item Prover selects a message $m$ from a message space $M$ which he wants to commit to:
+        $m \leftarrow \mathbb{M}$
+
+    \item Prover samples random value $r$ from a challange space $C$ (usually called blinding factor) from $\mathbb{Z}$:
+        $r \xleftarrow{R} \mathbb{C}$
+    
+    \item Both values will be concatenated and hashed with the hash function $H$ to produce the commitment:
+        $c = H(m \parallel r)$
+
+\end{enumerate}
+
+Commitment should be shared with a verifier. During the verification stage, prover reveals $(m, r)$ to the \textit{Verifier}. 
+To check the commitment, verifier computes: $c_1 = H(m \parallel r)$.
+
+If $c_1 = c$, prover has revealed the correct pair $(m, r)$.
+
+It should be noted that a cryptographic hash function aims to provide collision resistance, 
+meaning that the probability two different messages will result in one output is negligible.
+Because the \textit{Verifier} knows the hash function digest $c$ before the \textit{Prover} reveals $m$ and $r$, 
+the \textit{Prover} would need to find a collision $H(m' \parallel r') = H(m \parallel r)$ to be able to convince the \textit{Verifier}
+that $m'$ value was committed.
+
+However, due to the collision resistance, finding such $m'$ and $r'$ is computationally infeasible. Which means the \textit{Prover}
+won't be able to convince the \textit{Verifier} that the commitment was done to another value providing a \textit{binding} property.
+
+A cryptographically secure hash function is a one-way funcation, which means that finding the hash preimage is almost as hard as bruteforcing
+all possibile input values. Given large challenge space, the probability of the \textit{Verifier} of finding $(m, r)$ such that
+$H(m, r) = c$ is negligible, which ensures \textit{hiding} property of the commitment scheme. 
+
+
+\subsubsection{Pedersen commitments}
+
+Pedersen commitments allow us to represent arbitrarily large vectors with a single elliptic curve point, while optionally hiding any information about the vector. Pedersen commitment uses a public group $\mathbb{G}$ of order $q$ and two random public generators $G$ and $U$: $U = uG$. Secret parameter $u$ should be hidden, otherwise the $\textit{Binding}$ property of the commitment scheme will be violated.
```

Comment:
The secret parameter $u$ should be hidden; rephrase it a bit. It shouldn't be hidden; it must be unknown and unable to be reconstructed/recomputed in some way